### PR TITLE
Add don't pack option to Manual Bake

### PR DESCRIPTION
**Pull request by 989onan**
Maybe add a thing to not do this if decimating but keep it enabled if using normals because overlap is only a problem when normals and decimate are used together

### CODE REVIEW

**Comment by 989onan** [comment]
first comment has now been resolved on the auto disable part

**Comment by feilen** [comment]
I need to take a closer look, but would it be possible to add this as a different 'Generate UVMap' enum? I know it's not neccesarily correct in-name, but functionality wise it makes more sense to put it there.

If generate_uvmaps is getting too squished, you can remove 'expand=True' in ui/bake.py to turn it into a drop-down.

File: tools/bake.py [hunk 1]
```diff
@@ -718,6 +719,9 @@ def perform_bake(self, context):
         cleanup_shapekeys = context.scene.bake_cleanup_shapekeys # Reverted and _old shapekeys
         ignore_hidden = context.scene.bake_ignore_hidden
 
+        #adjusting don't pack to disable if there are things that will cause undesirable results if enabled, then invert and store to pack_uvs var
+        pack_uvs = not (dont_pack_uvs and (not emit_indirect) and (not prioritize_face) and (not pass_ao) and (not any(plat.use_decimation for plat in context.scene.bake_platforms)))
```
Comment (feilen) [inline tools/bake.py]: This should just be "not dont_pack_uvs", if someone has explicitly selected "don't pack" then we should honor that, even if it'll have bad results.

File: ui/bake.py [hunk 2]
```diff
@@ -320,6 +320,9 @@ def draw(self, context):
                     row = col.row(align=True)
                     row.separator()
                     row.prop(context.scene, 'bake_unwrap_angle', expand=True)
+                if context.scene.bake_uv_overlap_correction == "MANUALNOPACK" and any(plat.use_decimation for plat in context.scene.bake_platforms) and (not context.scene.bake_emit_indirect) and (not context.scene.bake_prioritize_face) and (not context.scene.bake_pass_ao):
+                    col.label(text="***Skipping the packing step will not work if one or more of your platforms has decimation active,***") #TRANSLATE PLEASE!
```
Comment (feilen) [inline ui/bake.py]: The emphasis is waaay overdone here. Just add a note that says:
"Warning: may have unexpected results if you have overlapping islands."

**Review by feilen — CHANGES_REQUESTED** [review]

File: ui/bake.py (hunk 2 above)
Comment (feilen) [reply ui/bake.py]: otherwise looking great! I meant to add something like this forever ago

**Comment by 989onan** [comment]
Fixed conflict with development branch so the codes are merged

**Comment by 989onan** [comment]
sorry this is messy, but I'm using the website since it won't clone to my computer. The branch is so borked on my end. Maybe after the development merge it won't break so much. Github is complaining that some of the dependencies on my version before I pulled development into it didn't exist anymore and it was angy

**Comment by Darkblader24** [comment]
Thank you very much!

**Pull request merged**